browser/brave_screenshots/brave_screenshots_tab_feature.cc

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/brave_screenshots/brave_screenshots_tab_feature.h"
+
+#include "base/notreached.h"
+#include "brave/browser/brave_screenshots/screenshots_utils.h"
+#include "brave/browser/brave_screenshots/strategies/fullpage_strategy.h"
+#include "brave/browser/brave_screenshots/strategies/selection_strategy.h"
+#include "brave/browser/brave_screenshots/strategies/viewport_strategy.h"
+#include "chrome/browser/image_editor/screenshot_flow.h"
+#include "chrome/browser/ui/browser.h"
+#include "content/public/browser/web_contents.h"
+
+namespace {
+
+// Some screenshots may need to be clipped to avoid the GPU limit.
+// See https://crbug.com/1260828. When this happens, we may wish to notify the
+// user that only a portion of their page could be captured.
+void DisplayScreenshotClippedNotification(
+    base::WeakPtr<content::WebContents> web_contents) {
+  // Issue: https://github.com/brave/brave-browser/issues/43369
+  NOTIMPLEMENTED();
+}
+
+}  // namespace
+namespace brave_screenshots {
+
+BraveScreenshotsTabFeature::BraveScreenshotsTabFeature() {
+  DVLOG(1) << "BraveScreenshotsTabFeature created";
+}
+
+BraveScreenshotsTabFeature::~BraveScreenshotsTabFeature() {
+  DVLOG(1) << "BraveScreenshotsTabFeature destroyed";
+}
+
+void BraveScreenshotsTabFeature::StartScreenshot(
+    base::WeakPtr<content::WebContents> web_contents,
+    ScreenshotType type) {
+  DVLOG(1) << "Called StartScreenshot";
+  CHECK(web_contents.get());
+
+  // Store the WebContents for later use
+  web_contents_ = web_contents;
```

Comment:
WebContents is fixed for the life of this class so it should be added in the constructor

Reply:
also there's no reason for it to be WeakPtr since the lifetime of the tab feature is tied to the webcontents